core/src/io/anuke/mindustry/ui/dialogs/CustomRulesDialog.java

```diff
@@ -1,67 +1,105 @@
 package io.anuke.mindustry.ui.dialogs;
 
+import io.anuke.arc.function.BooleanConsumer;
+import io.anuke.arc.function.BooleanProvider;
 import io.anuke.arc.function.FloatConsumer;
 import io.anuke.arc.function.FloatProvider;
+import io.anuke.arc.graphics.Color;
 import io.anuke.arc.scene.ui.layout.Table;
 import io.anuke.arc.util.Strings;
+import io.anuke.arc.util.Structs;
 import io.anuke.mindustry.core.Platform;
+import io.anuke.mindustry.game.Gamemode;
 import io.anuke.mindustry.game.Rules;
+import io.anuke.mindustry.graphics.Pal;
 
 import static io.anuke.mindustry.Vars.tilesize;
 
 public class CustomRulesDialog extends FloatingDialog{
-    public final Rules rules = new Rules();
     private Table main;
+    public Rules rules;
+    public Gamemode selectedGamemode;
 
     public CustomRulesDialog(){
         super("$mode.custom");
 
-        rules.waves = true;
-        rules.waveTimer = true;
-
         setFillParent(true);
         shown(this::setup);
         addCloseButton();
     }
 
+    public void show(Rules rules, Gamemode gamemode){this.rules = rules; this.selectedGamemode = gamemode; show();}
```

Comment:
Forgot to mention: These statements should be in separate lines. I only sent an example in Discord to give a general idea of how it would work.